src/isar/state_machine/states/send.py

```diff
@@ -26,70 +40,84 @@ def start(self):
 
     def stop(self):
         self.send_failure_counter = 0
+        if self.send_thread:
+            self.send_thread.join()
```

Comment:
Could we risk the code hanging on this step if something is holding up the thread?

Reply:
Well, that is the purpose of join come to think on it... What I mean is that is there a case where the code could hang indefinitely, if so, should we implement some sort of timeout?